### ISSUE TITLE
Discussion: Supporting accelerators like the Intel ACC100

### ISSUE THREAD

**Issue by DolceTriade** [k8snetworkplumbingwg/sriov-network-operator#197]
Hello,

Currently, accelerators like the ACC100 can be used to offload processing. It relies on DPDK like a lot of networking cards and its set up and flow is very similar to how the SRIOV network operator sets up NICs. Furthermore, the sriov-device-plugin supports accelerators as well. Theoretically, it should be possible to write a plugin that would initialize and manage an accelerator.

There exists an operator https://github.com/smart-edge-open/openshift-operator/tree/main/sriov-fec that does almost exactly the same things as the sriov-network-operator. Would it make sense to merge the functionality for uses cases that need to do both? Ultimately, both operators will end up writing the sriov-device-plugin config maps and it seems wasteful to have two of the same daemonset running on the same node.

However, with https://github.com/k8snetworkplumbingwg/sriov-network-operator/commit/1f42de54aaf46ccfb90396406d7ffa3dbf195906, devices that may support SR-IOV but do not have a network device associated with them will no longer be detected in the `SriovNetworkNodeStates` (like accelerators). 

Does supporting similar devices sound like something that should also be managed by the sriov-network-operator?

@mskrocki

**Comment by pliurh** [comment]
@DolceTriade What functions do you want the SRIOV network operator to configure the ACC100? creating VFs, loading drivers, and configuring the device plugin? I assume, there are some other functions of ACC100 that can not be managed by the SRIOV network operator. Do we still need another operator to manage those configurations? If yes, how do we coordinate between the 2 operators?

**Comment by DolceTriade** [comment]
Yes, that's the idea for what the sriov-network-operator would configure for the accelerator. 

The hope was that those other functions could be included in a plugin that could read another config CRD to handle that portion.
